### DISCUSSION TITLE
Switching mREPL to the package of the current file

### DISCUSSION THREAD

**Discussion by appblue** [joaotavora/sly#451] [Q&A] [answered]
Is there an option to switch mREPL to a package of the currently edited .lisp file - similar to cider-repl-set-ns functionality for Clojure/Cider tandem?

**Comment by joaotavora** [comment, marked as answer]
Try `C-c ~` or `M-x sly-mrepl-sync`.

**Reply by appblue** [reply]
Thanks for prompt reply. 

Actually, I have been able to find it in the docs, but it doesn't work when I run it on a file that is a part of my `local_projects` in quicklisp created by `cl-package:make-package` tool.

My mREPL is still in CL-USER package:

```
CL-USER> 
; Synched package to COMMON-LISP-USER and directory to /Users/u01/quicklisp-ccl/local-projects/l-banner/src/
CL-USER> 
; No values
CL-USER> *package*
#<Package "COMMON-LISP-USER">
```

**Reply by joaotavora** [reply]
I'd need to see the file.  Is its code loaded?  Can Sly recognize the package at point? If you answer "no" to any of these two questions, it won't work.

**Reply by appblue** [reply]
Fantastic - it works!!

 I think I was running `sly-mrepl-sync` with Emacs point set before `(in-package ...)`

**Reply by joaotavora** [reply]
Good to hear. Be sure to mark this question solved.

**Reply by joaotavora** [reply]
... if you can find the option to do so.  I can't find it, maybe because this isn't in the QA category

**Reply by appblue** [reply]
... I can't find it to be honest

**Reply by joaotavora** [reply]
If you reload you should be able, to.  If not, never mind

**Reply by AkashaP** [reply]
related question is there an easy way to switch back to cl-user? i basically want to evaluate (in-package :cl-user) as a hotkey but in the mrepl

**Reply by joaotavora** [reply]
@AkashaP if you want a dedicated hotkey for that, you probably have to program that in Emacs Lisp and put the function in your emacs.  Look in SLY code for examples.  Alternatively, to quickly switch back to cl-user, just type the `(in-package :cl-user)` once in the repl and then refer back to history with C-r and a few characters.That's what I do at least. Good luck.
